ts/index.ts

```diff
@@ -0,0 +1,216 @@
+
+  constructor() {
+    const options: ActionOptions = {
+      name: "flakehub-push",
+      fetchStyle: "nix-style",
+      diagnosticsUrl: new URL(
+        "https://install.determinate.systems/flakehub-push/telemetry",
+      ),
+      legacySourcePrefix: "flakehub-push",
+      requireNix: "ignore",
+    };
+
+    this.idslib = new IdsToolbox(options);
+
+    // Inputs
+    const visibility = this.verifyVisibility();
+    this.visibility = visibility;
+
+    this.name = inputs.getStringOrNull("name");
+    this.repository = inputs.getString("repository");
+    this.mirror = inputs.getBool("mirror");
+    this.directory = inputs.getString("directory");
+    this.gitRoot = inputs.getString("git-root");
+    this.tag = inputs.getString("tag");
+    this.rollingMinor = inputs.getNumberOrNull("rolling-minor");
+    this.rolling = inputs.getBool("rolling");
+    this.host = inputs.getString("host");
+    this.logDirectives = inputs.getString("log-directives");
+    this.logger = inputs.getString("logger");
+    this.gitHubToken = inputs.getString("github-token");
+    // extra-tags is deprecated but we still honor it
+    this.extraLabels =
+      inputs.getString("extra-labels") === ""
+        ? inputs.getString("extra-tags")
+        : "";
+    this.spdxExpression = inputs.getString("spdx-expression");
+    this.errorOnConflict = inputs.getBool("error-on-conflict");
+    this.includeOutputPaths = inputs.getBool("include-output-paths");
+    this.flakeHubPushBinary = inputs.getStringOrNull("flakehub-push-binary");
+  }
+
+  private verifyVisibility(): Visibility {
+    const visibility = inputs.getString("visibility");
+    if (!VISIBILITY_OPTIONS.includes(visibility)) {
+      throw new Error(
+        `Visibility option \`${visibility}\` not recognized. Available options: ${VISIBILITY_OPTIONS.join(", ")}.`,
+      );
+    }
+    return visibility as Visibility;
+  }
+
+  private async executionEnvironment(): Promise<ExecutionEnvironment> {
+    const env: ExecutionEnvironment = {};
+
+    env.FLAKEHUB_PUSH_VISIBLITY = this.visibility;
+    env.FLAKEHUB_PUSH_TAG = this.tag;
+    env.FLAKEHUB_PUSH_HOST = this.host;
+    env.FLAKEHUB_PUSH_LOG_DIRECTIVES = this.logDirectives;
+    env.FLAKEHUB_PUSH_LOGGER = this.logger;
+    env.FLAKEHUB_PUSH_GITHUB_TOKEN = this.gitHubToken;
+    env.FLAKEHUB_PUSH_NAME = this.flakeName;
+    env.FLAKEHUB_PUSH_REPOSITORY = this.repository;
+    env.FLAKEHUB_PUSH_DIRECTORY = this.directory;
+    env.FLAKEHUB_PUSH_GIT_ROOT = this.gitRoot;
+    env.FLAKEHUB_PUSH_EXTRA_LABELS = this.extraLabels;
+    env.FLAKEHUB_PUSH_SPDX_EXPRESSION = this.spdxExpression;
+
+    if (!this.rolling) {
+      env.FLAKEHUB_PUSH_ROLLING = "false";
+    }
+
+    if (!this.mirror) {
+      env.FLAKEHUB_PUSH_MIRROR = "false";
+    }
+
+    if (!this.errorOnConflict) {
+      env.FLAKEHUB_PUSH_ERROR_ON_CONFLICT = "false";
+    }
+
+    if (!this.includeOutputPaths) {
+      env.FLAKEHUB_PUSH_INCLUDE_OUTPUT_PATHS = "false";
+    }
```

Comment:
I'm wondering if the more important cases are setting these to true than setting this to false, but why not set it in either case?